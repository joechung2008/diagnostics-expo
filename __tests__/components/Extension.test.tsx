import { render } from '@testing-library/react-native';
import Extension from '../../components/Extension';

// Mock the ThemeContext
jest.mock('../../components/ThemeContext', () => ({
  useTheme: () => ({
    colors: {
      background: '#ffffff',
      surface: '#f5f5f5',
      text: '#000000',
      tableBorder: '#cccccc',
      tableHeaderBg: '#e0e0e0',
    },
  }),
}));

// Mock child components
jest.mock('../../components/Configuration', () => {
  return {
    __esModule: true,
    default: 'MockConfiguration',
  };
});

jest.mock('../../components/StageDefinition', () => {
  return {
    __esModule: true,
    default: 'MockStageDefinition',
  };
});

describe('Extension', () => {
  const mockConfig = {
    apiUrl: 'https://api.example.com',
    timeout: '30000',
  };

  const mockStageDefinition = {
    development: ['dev-server-1', 'dev-server-2'],
    production: ['prod-server-1'],
  };

  it('renders extension name and config', () => {
    const { toJSON } = render(
      <Extension
        extensionName="Test Extension"
        config={mockConfig}
        manageSdpEnabled={true}
      />
    );
    expect(toJSON()).toMatchSnapshot();
  });

  it('renders extension name and stage definition', () => {
    const { toJSON } = render(
      <Extension
        extensionName="Another Extension"
        stageDefinition={mockStageDefinition}
        manageSdpEnabled={false}
      />
    );
    expect(toJSON()).toMatchSnapshot();
  });

  it('renders extension name with both config and stage definition', () => {
    const { toJSON } = render(
      <Extension
        extensionName="Full Extension"
        config={mockConfig}
        stageDefinition={mockStageDefinition}
        manageSdpEnabled={true}
      />
    );
    expect(toJSON()).toMatchSnapshot();
  });

  it('renders extension name only when no config or stage definition', () => {
    const { toJSON } = render(
      <Extension extensionName="Minimal Extension" manageSdpEnabled={false} />
    );
    expect(toJSON()).toMatchSnapshot();
  });
});
