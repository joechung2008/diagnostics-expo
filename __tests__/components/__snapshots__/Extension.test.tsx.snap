// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Extension renders extension name and config 1`] = `
<RCTScrollView
  contentContainerStyle={
    {
      "display": "flex",
      "flexDirection": "column",
      "gap": 16,
      "padding": 10,
    }
  }
  style={
    {
      "backgroundColor": "#ffffff",
      "flex": 1,
    }
  }
>
  <View>
    <Text
      style={
        {
          "color": "#000000",
          "fontSize": 18,
          "fontWeight": "600",
          "marginBottom": 10,
        }
      }
    >
      Test Extension
    </Text>
    <MockConfiguration
      config={
        {
          "apiUrl": "https://api.example.com",
          "timeout": "30000",
        }
      }
    />
  </View>
</RCTScrollView>
`;

exports[`Extension renders extension name and stage definition 1`] = `
<RCTScrollView
  contentContainerStyle={
    {
      "display": "flex",
      "flexDirection": "column",
      "gap": 16,
      "padding": 10,
    }
  }
  style={
    {
      "backgroundColor": "#ffffff",
      "flex": 1,
    }
  }
>
  <View>
    <Text
      style={
        {
          "color": "#000000",
          "fontSize": 18,
          "fontWeight": "600",
          "marginBottom": 10,
        }
      }
    >
      Another Extension
    </Text>
    <MockStageDefinition
      stageDefinition={
        {
          "development": [
            "dev-server-1",
            "dev-server-2",
          ],
          "production": [
            "prod-server-1",
          ],
        }
      }
    />
  </View>
</RCTScrollView>
`;

exports[`Extension renders extension name only when no config or stage definition 1`] = `
<RCTScrollView
  contentContainerStyle={
    {
      "display": "flex",
      "flexDirection": "column",
      "gap": 16,
      "padding": 10,
    }
  }
  style={
    {
      "backgroundColor": "#ffffff",
      "flex": 1,
    }
  }
>
  <View>
    <Text
      style={
        {
          "color": "#000000",
          "fontSize": 18,
          "fontWeight": "600",
          "marginBottom": 10,
        }
      }
    >
      Minimal Extension
    </Text>
  </View>
</RCTScrollView>
`;

exports[`Extension renders extension name with both config and stage definition 1`] = `
<RCTScrollView
  contentContainerStyle={
    {
      "display": "flex",
      "flexDirection": "column",
      "gap": 16,
      "padding": 10,
    }
  }
  style={
    {
      "backgroundColor": "#ffffff",
      "flex": 1,
    }
  }
>
  <View>
    <Text
      style={
        {
          "color": "#000000",
          "fontSize": 18,
          "fontWeight": "600",
          "marginBottom": 10,
        }
      }
    >
      Full Extension
    </Text>
    <MockConfiguration
      config={
        {
          "apiUrl": "https://api.example.com",
          "timeout": "30000",
        }
      }
    />
    <MockStageDefinition
      stageDefinition={
        {
          "development": [
            "dev-server-1",
            "dev-server-2",
          ],
          "production": [
            "prod-server-1",
          ],
        }
      }
    />
  </View>
</RCTScrollView>
`;
